src/cpr_data_access/models.py

```diff
@@ -43,6 +43,64 @@
 AnyDocument = TypeVar("AnyDocument", bound="BaseDocument")
 
 
+def _load_and_validate_metadata_csv(
+    metadata_csv_path: Path, target_model: type[AnyDocument]
+) -> pd.DataFrame:
+    """Load a metadata CSV, raising a ValueError if it does not exist or doesn't have the expected columns."""
+    if not metadata_csv_path.exists():
+        raise ValueError(f"metadata_csv_path {metadata_csv_path} does not exist")
+
+    if not metadata_csv_path.is_file() or not metadata_csv_path.suffix == ".csv":
+        raise ValueError(f"metadata_csv_path {metadata_csv_path} must be a csv file")
+
+    metadata_df = pd.read_csv(metadata_csv_path)
+
+    expected_cols = {
+        "Geography",
+        "Geography ISO",
+        "CPR Document Slug",
+        "Category",
+        "CPR Collection ID",
+        "CPR Family ID",
+        "CPR Family Slug",
+        "CPR Document Status",
+    }
+
+    cclw_expected_cols = {
+        "Sectors",
+        "Collection name",
+        "Document Type",
+        "Family name",
+        "Document role",
+        "Document variant",
+    }
+
+    gst_expected_cols = {
+        "Author",
+        "Author Type",
+        "Date",
+        "Documents",  # URL
```

Comment:
Do you agree that for future csv' we should enforce this as one source url per row? 

This was a bit of a pain in the ascor ingest having source urls separated by new lines. I don't think as well that we have a way in our product data model of storing multiple source urls for a document (just in case we did ingest this data)

Reply:
yeah agreed – i didn't realise that was the case for ASCOR. this is only intended for gst and cpr core csvs which should obey this